CRM/Activity/BAO/Activity.php

```diff
@@ -317,6 +317,8 @@ public static function create(&$params) {
 
     // CRM-9137
     if (!empty($params['id'])) {
+      $activity = CRM_Activity_DAO_Activity::findById($params['id']);
+
       CRM_Utils_Hook::pre('edit', 'Activity', $activity->id, $params);
```

Comment:
@dejan9393 Instead of loading the whole object into `$activity` and passing id - can we not pass id directly to the pre() hook. Similar to -

    CRM_Utils_Hook::pre('edit', 'Activity', $params['id'], $params);

If yes, can you please modify with the same. 

Thanks.